### ISSUE TITLE
UnicodeEecodeError while running hypothesis

### ISSUE THREAD

**Issue by thedrow** [pytest-dev/pytest#1880]
This happened to me on pytest 3.0.1 running on OSX El Capitan with Python 2.7.
I have no idea how to reproduce this but I think that we shouldn't be calling str() there on Python 2.

Here is what I have installed:
```
pip list
apipkg (1.4)
appnope (0.1.0)
backports.shutil-get-terminal-size (1.0.0)
cffi (1.7.0)
decorator (4.0.10)
docutils (0.12)
enum34 (1.1.6)
execnet (1.4.1)
greenlet (0.4.9)
hashids (1.1.0)
hashids-cffi (0.1.0, /Users/omer.katz/Documents/hashids)
hiredis (0.1.0, /Users/omer.katz/Documents/hashids)
hypothesis (3.4.2)
ipython (5.1.0)
ipython-genutils (0.1.0)
pathlib (1.0.1)
pathlib2 (2.1.0)
pexpect (4.2.1)
pickleshare (0.7.4)
pip (8.1.2)
pluggy (0.3.1)
prompt-toolkit (1.0.7)
ptyprocess (0.5.1)
py (1.4.31)
Pygments (2.1.3)
pytest (3.0.1)
pytest-benchmark (3.0.0)
pytest-xdist (1.15.0)
readline (6.2.4.1)
setuptools (26.0.0)
simplegeneric (0.8.1)
six (1.10.0)
statistics (1.0.3.5)
tox (2.3.1)
traitlets (4.2.2)
virtualenv (15.0.3)
wcwidth (0.1.7)
wheel (0.29.0)
```

```
tests/test_hypothesis.py ....Falsifying example: test_decode_unicode_hashid(salt=u'', alphabet='acdefghijklmnobp', min_length=0, hashid=u'\u0bb8')

INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/main.py", line 96, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/main.py", line 131, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/vendored_packages/pluggy.py", line 596, in execute
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/main.py", line 152, in pytest_runtestloop
INTERNALERROR>     item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
INTERNALERROR>     return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
INTERNALERROR>     _MultiCall(methods, kwargs, hook.spec_opts).execute()
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
INTERNALERROR>     return _wrapped_call(hook_impl.function(*args), self.execute)
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/vendored_packages/pluggy.py", line 253, in _wrapped_call
INTERNALERROR>     return call_outcome.get_result()
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/vendored_packages/pluggy.py", line 279, in get_result
INTERNALERROR>     _reraise(*ex)  # noqa
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/vendored_packages/pluggy.py", line 264, in __init__
INTERNALERROR>     self.result = func()
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/vendored_packages/pluggy.py", line 596, in execute
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/runner.py", line 66, in pytest_runtest_protocol
INTERNALERROR>     runtestprotocol(item, nextitem=nextitem)
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/runner.py", line 79, in runtestprotocol
INTERNALERROR>     reports.append(call_and_report(item, "call", log))
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/runner.py", line 133, in call_and_report
INTERNALERROR>     call = call_runtest_hook(item, when, **kwds)
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/runner.py", line 151, in call_runtest_hook
INTERNALERROR>     return CallInfo(lambda: ihook(item=item, **kwds), when=when)
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/runner.py", line 168, in __init__
INTERNALERROR>     self.excinfo = ExceptionInfo()
INTERNALERROR>   File "/Users/omer.katz/.virtualenvs/hashids-cffi/site-packages/_pytest/_code/code.py", line 357, in __init__
INTERNALERROR>     exprinfo = str(tup[1])
INTERNALERROR> UnicodeEncodeError: 'ascii' codec can't encode character u'\u0bb8' in position 117: ordinal not in range(128)
```

**Comment by The-Compiler** [comment]
Can you check if this is fixed with the latest `master` (via #1870)?

**Comment by thedrow** [comment]
I'm not sure if I can reproduce this since the test input is random.

**Comment by The-Compiler** [comment]
hypothesis will save failing examples in its database and then make sure to retry them when you run it again.

**Comment by thedrow** [comment]
I'll give it a shot :)

**Comment by nicoddemus** [comment]
I'm closing this for now as I'm pretty confident this has been fixed by #1870, but please @thedrow don't hesitate on re-opening this issue if that's not the case. :grin:

**Comment by thedrow** [comment]
Yup. Fixed :)
